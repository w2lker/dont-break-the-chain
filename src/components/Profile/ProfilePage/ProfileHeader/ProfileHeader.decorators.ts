import withStyles from '@material-ui/core/styles/withStyles';
import { Dispatch } from 'react';
import { connect } from 'react-redux';

import ProfileHeader from './ProfileHeader';
import profileHeaderStyles from './ProfileHeader.styles';

import { logout } from '../../../../actions/auth';

const mapStateToProps = null;
const mapDispatchToProps = (dispatch: Dispatch<any>) => {
  return {
    logout: () => dispatch(logout()),
  };
};

// @ts-ignore
const connectedComponent = connect(mapStateToProps, mapDispatchToProps)(ProfileHeader);


const styledComponent = withStyles(profileHeaderStyles)(connectedComponent);

export default styledComponent;
