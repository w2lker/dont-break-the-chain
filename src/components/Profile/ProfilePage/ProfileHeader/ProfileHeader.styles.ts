import { Styles } from '@material-ui/styles/withStyles';
import { bon } from '../../../../utils/styleUtils';

const profileHeaderStyles: Styles<{}, {}, any> = {
  root: {
    position: 'relative',
    marginBottom: 5,
    textAlign: 'center',
    boxShadow: '0px 4px 8px rgba(176, 190, 197, 0.24)',
  },
  title: {
    ...bon.setFont(20, '#263238', 500, '0.15px'),
    margin: 0,
    padding: '15px 0',
    lineHeight: '28px',
  },
  button: {
    ...bon.positionMirrored('absolute', '4px 0 unset unset'),
    ...bon.size(52),
  },
};

export default profileHeaderStyles;
