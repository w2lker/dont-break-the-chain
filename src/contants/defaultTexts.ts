const defaultTexts = {
  navigationMenu: {
    chains: 'Chains',
    habits: 'Habits',
    profile: 'Profile',
  },
  profile: {
    header: {
      title: 'Profile',
    },
  },
};

export default defaultTexts;
