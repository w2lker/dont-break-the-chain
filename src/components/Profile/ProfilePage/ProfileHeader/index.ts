import ProfileHeader from './ProfileHeader.decorators';

export default ProfileHeader;
