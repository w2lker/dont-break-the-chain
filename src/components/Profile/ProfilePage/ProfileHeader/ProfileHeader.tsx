import * as React from 'react';

import { IconButton } from '@material-ui/core';
import defaultTexts from '../../../../contants/defaultTexts';

import MaterialDesignIcon from '../../../basic/Material-Icon';

export interface IProfileHeaderProps {
  classes?: any;
  logout?: () => void;
}

const ProfileHeader: React.FC<IProfileHeaderProps> = (props) => {
  const { classes, logout } = props;
  const texts = defaultTexts.profile.header;
  return (
    <div className={classes.root}>
      <h3 className={classes.title}>
        {texts.title}
      </h3>
      <IconButton
        className={classes.button}
        onClick={logout}
      >
        <MaterialDesignIcon
          name="logout"
          className="header-icon"
        />
      </IconButton>
    </div>
  );
};

ProfileHeader.defaultProps = {
  classes: {
    root: '',
    title: '',
    button: '',
  },
};

export default ProfileHeader;
