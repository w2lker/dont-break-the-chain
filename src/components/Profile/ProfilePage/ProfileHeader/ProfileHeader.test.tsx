import * as React from 'react';

import { mount, shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import defaultTexts from '../../../../contants/defaultTexts';

import ProfileHeader, { IProfileHeaderProps } from './ProfileHeader';
import ProfileHeaderDecorated from './ProfileHeader.decorators';
import profileHeaderStyles from './ProfileHeader.styles';

const classes = {
  root: 'profile-header-root',
  title: 'profile-header-title',
  button: 'profile-header-button',
};
const classesKeys = Object.keys(classes);

describe('Profile Header Component', () => {
  const props: IProfileHeaderProps = {
    classes,
    logout: jest.fn(),
  };
  it('match snapshot', () => {
    const component = mount(<ProfileHeader {...props} />);
    expect(component.debug()).toMatchSnapshot();
  });
  it('renders', () => {
    const component = shallow(<ProfileHeader {...props} />);
    // It renders wrapper with root class
    const wrapper = component.find(`div.${classes.root}`);
    expect(wrapper.length).toBe(1);
    // It renders title with title class and text
    const title = component.find(`h3.${classes.title}`);
    expect(title.length).toBe(1);
    expect(title.text()).toBe(defaultTexts.profile.header.title);
    // It renders icon button with class and passes onClick action
    const button = component.find(`.${classes.button}`);
    expect(button.length).toBe(1);
    expect(button.hasClass(classes.button)).toBeTruthy();
    expect(button.props().onClick).toBeTruthy();
  });

  it('executes logout on click', () => {
    const component = shallow(<ProfileHeader {...props} />);
    const button = component.find(`.${classes.button}`);
    button.simulate('click');
    expect(props.logout).toBeCalledTimes(1);
  });
});

describe('Profile Header Styles', () => {
  it('should contain principle classes', () => {
    expect(profileHeaderStyles).toBeDefined();
    classesKeys.forEach((keyValue) => {
      // @ts-ignore
      expect(profileHeaderStyles[keyValue]).toBeDefined();
    });
  });
});

describe('Profile Header Decorated', () => {
  const middleWares = [thunk];
  const mockStore = configureMockStore(middleWares);

  const store = mockStore({});
  const component = mount(
    <Provider store = {store}>
      < ProfileHeaderDecorated />);
    </Provider>,
  );

  it('provides styled classes from decorators', () => {
    // @ts-ignore
    const assignedClasses = component.find('ProfileHeader').props().classes;
    classesKeys.forEach((keyValue) => {
        // @ts-ignore
      expect(assignedClasses[keyValue]).toBeDefined();
    });
  });

  it('provides dispatch function to logout', () => {
    // @ts-ignore
    const logout = component.find('ProfileHeader').props().logout;
    expect(typeof logout).toBe('function');
  });

});
