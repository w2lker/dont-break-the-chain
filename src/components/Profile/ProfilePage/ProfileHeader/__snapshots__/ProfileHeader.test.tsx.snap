// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Header Component match snapshot 1`] = `
"<ProfileHeader classes={{...}} logout={[Function: mockConstructor]}>
  <div className=\\"profile-header-root\\">
    <h3 className=\\"profile-header-title\\">
      Profile
    </h3>
    <WithStyles(ForwardRef(IconButton)) className=\\"profile-header-button\\" onClick={[Function: mockConstructor]}>
      <ForwardRef(IconButton) classes={{...}} className=\\"profile-header-button\\" onClick={[Function: mockConstructor]}>
        <WithStyles(ForwardRef(ButtonBase)) className=\\"MuiIconButton-root profile-header-button\\" centerRipple={true} focusRipple={true} disabled={false} onClick={[Function: mockConstructor]}>
          <ForwardRef(ButtonBase) classes={{...}} className=\\"MuiIconButton-root profile-header-button\\" centerRipple={true} focusRipple={true} disabled={false} onClick={[Function: mockConstructor]}>
            <button className=\\"MuiButtonBase-root MuiIconButton-root profile-header-button\\" onBlur={[Function]} onClick={[Function: mockConstructor]} onFocus={[Function]} onKeyDown={[Function]} onKeyUp={[Function]} onMouseDown={[Function]} onMouseLeave={[Function]} onMouseUp={[Function]} onDragLeave={[Function]} onTouchEnd={[Function]} onTouchMove={[Function]} onTouchStart={[Function]} tabIndex={0} type=\\"button\\" disabled={false}>
              <span className=\\"MuiIconButton-label\\">
                <MaterialDesignIcon name=\\"logout\\" className=\\"header-icon\\">
                  <span className=\\"material-icon mdi mdi-logout header-icon\\" style={[undefined]} />
                </MaterialDesignIcon>
              </span>
              <NoSsr>
                <WithStyles(undefined) center={true}>
                  <ForwardRef(TouchRipple) classes={{...}} center={true}>
                    <span className=\\"MuiTouchRipple-root\\">
                      <TransitionGroup component={{...}} exit={true} childFactory={[Function: childFactory]} />
                    </span>
                  </ForwardRef(TouchRipple)>
                </WithStyles(undefined)>
              </NoSsr>
            </button>
          </ForwardRef(ButtonBase)>
        </WithStyles(ForwardRef(ButtonBase))>
      </ForwardRef(IconButton)>
    </WithStyles(ForwardRef(IconButton))>
  </div>
</ProfileHeader>"
`;
